pkg/flexvolume/rbd/flexvolume.go

```diff
@@ -0,0 +1,267 @@
+}
+
+// Invocation: <driver executable> detach <mount device> <node name>
+func (d *FlexVolumeDriver) detach(mountDev, nodeName string) (map[string]interface{}, error) {
+	return nil, nil
+}
+
+// Invocation: <driver executable> waitforattach <mount device> <json options>
+func (d *FlexVolumeDriver) waitForAttach(mountDev, jsonOptions string) (map[string]interface{}, error) {
+	return map[string]interface{}{"device": mountDev}, nil
+}
+
+// Invocation: <driver executable> isattached <json options> <node name>
+func (d *FlexVolumeDriver) isAttached(jsonOptions, nodeName string) (map[string]interface{}, error) {
+	return map[string]interface{}{"attached": true}, nil
+}
+
+// Invocation: <driver executable> mount <mount dir> <json options>
+// mount persist meta data generated from jsonOptions into a tag file in target dir.
+func (d *FlexVolumeDriver) mount(targetMountDir, jsonOptions string) (map[string]interface{}, error) {
+	if err := d.initFlexVolumeDriverForMount(jsonOptions); err != nil {
+		return nil, err
+	}
+
+	// Step 1: Format the device.
+	if err := formatDisk(d.volId, d.pool, d.fsType); err != nil {
+		return nil, err
+	}
+
+	// Step 2: Create a json file and write metadata into the it.
+	data, err := d.generateOptionsData()
+	if err != nil {
+		return nil, err
+	}
+
+	optsData := &flexvolume.FlexVolumeOptsData{
+		CephRBDData: data,
+	}
+
+	if err := flexvolume.WriteJsonOptsFile(targetMountDir, optsData); err != nil {
+		os.Remove(targetMountDir)
+		return nil, err
+	}
+
+	glog.V(5).Infof("[Mount] Ceph RBD tag file is created in: %s with data: %s", targetMountDir, optsData)
+
+	return nil, nil
+}
+
+// generateOptionsData generates metadata for given ceph rbd volume.
+func (d *FlexVolumeDriver) generateOptionsData() (*flexvolume.CephRBDOptsData, error) {
+	user, keyring, monitors, err := readCephConfig(flexvolume.CephRBDConfigFile, flexvolume.CephRBDKeyringFile)
+	if err != nil {
+		return nil, err
+	}
+
+	return &flexvolume.CephRBDOptsData{
+		VolumeID: d.volId,
+		Pool:     d.pool,
+		FsType:   d.fsType,
+		User:     user,
+		Keyring:  keyring,
+		Monitors: monitors,
+	}, nil
+}
+
+// Invocation: <driver executable> unmount <mount dir>
+func (d *FlexVolumeDriver) unmount(targetMountDir string) (map[string]interface{}, error) {
+	// check the target directory
+	if _, err := os.Stat(targetMountDir); os.IsNotExist(err) {
+		return nil, fmt.Errorf("volume directory: %v does not exist", targetMountDir)
+	}
+
+	// NOTE: the targetDir will be cleaned by flexvolume,
+	// we just need to clean up the metadata file.
+	if err := flexvolume.CleanUpMetadataFile(targetMountDir); err != nil {
+		return nil, err
+	}
+
+	glog.V(5).Infof("[Unmount] Ceph RBD umounted: %s, and volume folder been cleaned: %s", d.volId, targetMountDir)
+
+	return nil, nil
+}
+
+type driverOp func(*FlexVolumeDriver, []string) (map[string]interface{}, error)
+
+type cmdInfo struct {
+	numArgs int
+	run     driverOp
+}
+
+var commands = map[string]cmdInfo{
```

Comment:
Is it possible to share this part of common code below?

Reply:
Done.